xlml/utils/bigquery.py

```diff
@@ -110,6 +108,47 @@ def is_valid_metric(self, value: float):
     invalid_values = [math.inf, -math.inf, math.nan]
     return not (value in invalid_values or math.isnan(value))
 
+  def delete(self, row_ids: Iterable[str]) -> None:
+    """Delete records from tables.
+
+    There is a known issue that you cannot delete or update over table
```

Comment:
Concretely, what sort of errors would this cause in our case? Duplicate entries?

Reply:
Yeah, at high level, in prod stage, I think it should only produce 1 metric record for an execution (i.e. a dag is scheduled on daily basis, and there should only have 1 record from day 2024-1-4, and 1 record from day 2024-1-5 etc. In the current implementation, if we clear the old execution, and a new metric record will be inserted into the BigQuery table. Please see more context in b/318896353.

Reply:
Discussed offline. Will go through the approach below to avoid data deletion and 90 min buffer:

* Update uuid to be unique for a specific execution on Airflow
* Create a view table for executions with the latest timestamp: so duplicates are still available in original datasets and view table only has 1 record for each exectuion

Reply:
After a second thought, we could create that view table for job_history as it has timestamp, but we cannot do that for metric_history and metadata_history. 

Schema for metric_history: job_uuid, metric_key, and metric_value
Schema for metadata_history: job_uuid, metadata_key, and metadata_value